##A brief Anthill tutorial with test data

This worked out example will take you through the basics of using the anthill pipeline. If any dependencies need to be installed which are all pointed out in the readme, first head to the "Dependency Installation Walkthrough."

Now, select your data. We have chosen .bam files from the compbio server for this purpose. These .bam files can be found [here](https://www.ebi.ac.uk/arrayexpress/experiments/E-GEUV-1/). Within this test_example directory, you will find a text file called example_bams.txt. Opening it shows the following .bam files:  
<details>
<summary>.bam files</summary>
<p>
HG00133.2.M_111216_2.bam
HG00324.1.M_111124_2.bam
HG00326.1.M_111124_7.bam
NA07051.5.M_120131_1.bam
NA19209.6.M_120217_1.bam
NA20519.1.M_111124_5.bam
</p>
</details>

If you are on the compbio server, these files are readily available within the directory `/homes/hwheeler/Data/gEUVADIS_RNASeq` 
You can copy them all to your directory using:

`while read line; do cp /homes/hwheeler/Data/gEUVADIS_RNASeq/$line /path/to/directory; done < example_bams.txt`

__Note__: For best results it is best to place all scripts within the same directory as the .bam data, not in an adjacent, sub, or parent directory. This has been done for you.

We now want to calculate intron excision ratios run the leaf-command script. To execute the script, use the location of your bamfiles and the location of your leafcutter install directory as arguments
`./leaf-command /path/to/bamfiles /path/to/leafcutter`

#Note:This step can take roughly an hour for this example, even longer with additional files. If you do not wish to wait around and keep your connection active it is recommended that you use the `nohup` command

Next, we want to parse our VCF files which contain the SNP information. Parsing any VCF file to output the genotype information takes some time we have provided a parsed example for you in SNPGenotypes_chr22 & SNPLoc_22.

SNP genotype files need to be further processed to match the gene expression files generated by leafcutter
The general syntax for this is as follows: 

`python file_order.py /path/to/SNPGenotypeFile /path/to/GeneExpressionFile`

The file will output as `ReorderedSNPGenotypes_chr22`

The ReorderedSNPGenotype_chr22 file will have an extra column that must be removed:
`cut -f 2- ReorderedSNPGenotype_chr22 > finalSNPGenotypes_chr22`

Now, we can run Matrix eQTL:
`Rscript matrix_eQTL.R /path/to/test_data/ /path/to/finalSNPGenotypes_chrX /path/from/test_data/SNPLoc_chrX
/path/to/test_data/Gene_expression_chrX /path/to/test_data/Gene_location_chrX`

`Rscript matrix_eQTL.R /path/to/test_data/ finalSNPGenotypes_chr22 SNPLoc_chr22 Gene_expression_chr22 Gene_location_chr22
